src/cmd_options.cpp

```diff
@@ -1,11 +1,94 @@
 #include "cmd_options.h"
+#include <array>
+#include <format>
+#include <utility>
 
 namespace CryptoGuard {
 
-ProgramOptions::ProgramOptions() : desc_("Allowed options") {}
+ProgramOptions::ProgramOptions() : desc_("Allowed options"), command_(ProgramOptions::COMMAND_TYPE::HELP) {
 
-ProgramOptions::~ProgramOptions() = default;
+    namespace po = boost::program_options;
+    desc_.add_options()("help", "print help message")("command", po::value<std::string>(),
+                                                      "command to execute: encrypt/decrypt/checksum")(
+        "input", po::value<std::string>(), "input file path")("output", po::value<std::string>(), "output file path")(
+        "password", po::value<std::string>(), "owner password");
+}
 
-bool ProgramOptions::Parse(int argc, char *argv[]) { return false; }
+std::pair<bool, std::string> ProgramOptions::Parse(int ac, char *av[]) {
+    auto vm = parseCommandLine(ac, av);
+
+    if (vm.contains("help")) {
+        return {true, ""};
+    }
```

Comment:
Ошибка появляется при использовании `notify`, поэтому команду help можно проверить в самом начале. С таким кодом require будет работать
```suggestion
    if (vm.contains("help")) {
        return {true, ""};
    }
    boost::program_options::notify(vm);

```

Reply:
Спасибо, теперь разобрался.